src/Lifecycle.js

```diff
@@ -128,6 +128,7 @@ function _loginWithToken(queryParams) {
         console.log("Logged in with token");
         setLoggedIn({
             userId: data.user_id,
+            deviceId: data.device_id,
```

Comment:
Does this definitely work - I have a feeling going through localstorage will cause your `undefined` to turn into  the string `"undefined"`.